### PR TITLE
Clarify custom-graph multiprocessing example

### LINKED ISSUE #9438: Clarify custom-graph multiprocessing example
```
import dask
from dask.multiprocessing import get
import numpy as np


def load(filename):
    return np.random.random(32, 4)


def clean(data):
    data[:, 0] = 0.1
    return data


def analyze(sequence_of_data):
    out = sequence_of_data[0]
    for a in sequence_of_data[1:]:
        out += a
    return out


def store(result):
    with open(..., "w") as f:
        f.write(result)


dsk = {
    "load-1": (load, "myfile.a.data"),
    "load-2": (load, "myfile.b.data"),
    "load-3": (load, "myfile.c.data"),
    "clean-1": (clean, "load-1"),
    "clean-2": (clean, "load-2"),
    "clean-3": (clean, "load-3"),
    "analyze": (analyze, ["clean-%d" % i for i in [1, 2, 3]]),
    "store": (store, "analyze"),
}

get(dsk, "store")  # executes in parallel

```

error:

```
lib/python3.9/multiprocessing/spawn.py", line 134, in _check_not_importing_main
    raise RuntimeError('''
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.

        This probably means that you are not using fork to start your
        child processes and you have forgotten to use the proper idiom
        in the main module:

            if __name__ == '__main__':
                freeze_support()
                ...

        The "freeze_support()" line can be omitted if the program
        is not going to be frozen to produce an executable.
Traceback (most recent call last):
```

### PR DESCRIPTION
**Pull request by noumxn**
While multiprocessing, if a notebook is not being used, it is required that the `get` command in the given example needs to be run inside a `if __name__ == '__main__':` block. Added clarification regarding this in the documentation.

- [x] Closes #9438 
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

### CODE REVIEW

**Comment by GPUtester** [comment]
Can one of the admins verify this patch?

Admins can comment `ok to test` to allow this one PR to run or `add to allowlist` to allow all future PRs from the same author to run.

File: docs/source/custom-graphs.rst [hunk 1]
```diff
@@ -55,6 +55,15 @@ analyze pipeline:
...
    get(dsk, 'store')  # executes in parallel
 
+In the above example, while multiprocessing, if a notebook is not being 
+used to run, the ``get(dsk, 'store')`` command needs to be run inside an 
+``if __name__ == '__main__':`` block as shown below:
+
+.. code-block:: python
+
+   if __name__ == '__main__':
+      get(dsk, "store")
```
Comment (rjzamora) [inline docs/source/custom-graphs.rst:65]:
```suggestion
If the above example is not being run inside a notebook, the ``get(dsk, 'store')``
command should be placed inside an ``if __name__ == '__main__':`` block:

.. code-block:: python

   if __name__ == '__main__':
       get(dsk, "store")
       
For an explanation of why ``if __name__ == '__main__'`` is necessary, see the
standard `multiprocessing guidelines <https://docs.python.org/3/library/multiprocessing.html#programming-guidelines>`_
for Python.
```

What do you think about this revision? (Your version is good - just adding slightly more context)

Comment (noumxn) [reply docs/source/custom-graphs.rst]: Looks great!!

**Review by rjzamora — APPROVED** [review]

**Comment by rjzamora** [comment]
Thanks @noumxn!

**Comment by jrbourbeau** [comment]
Thanks @noumxn @rjzamora. Looking at this example, it doesn't appear that using the `multiprocessing` scheduler is meaningful. Thoughts on just switching to `threads` instead?

**Comment by rjzamora** [comment]
>it doesn't appear that using the multiprocessing scheduler is meaningful. Thoughts on just switching to threads instead?

I agree that the original example should probably be using `threads` for simplicity. However, I do think it makes sense to keep the small note about adding the `__main__` block for multiprocessing.

**Comment by jrbourbeau** [comment]
> However, I do think it makes sense to keep the small note about adding the __main__ block for multiprocessing

Fair enough. Is this the right place for this note though? Needing to add `if __name__ == "__main__":` comes up enough (and also applies the `LocalCluster`) that we might want to have it somewhere more prominent than the page on custom graphs.

**Comment by rjzamora** [comment]
>Is this the right place for this note though?

No, I agree that the note should be in a more general place, and just be referenced in places where examples require `__main__` outside of a notebook (which doesn't need to include the example highlighted here if we move to `threaded`).

**Comment by jrbourbeau** [comment]
I'll push something up for this

**Comment by jrbourbeau** [comment]
See https://github.com/dask/dask/pull/9513

**Pull request merged**